device/cluster.cpp

```diff
@@ -452,15 +452,16 @@ std::unique_ptr<Chip> Cluster::construct_chip_from_cluster(
     }
 }
 
-std::unique_ptr<Chip> Cluster::construct_chip_from_cluster(chip_id_t chip_id, tt_ClusterDescriptor* cluster_desc) {
+std::unique_ptr<Chip> Cluster::construct_chip_from_cluster(
+    chip_id_t chip_id,
+    tt_ClusterDescriptor* cluster_desc,
+    bool perform_harvesting,
```

Comment:
I think it would just be nice to document what are the expected use cases for `perform_harvesting = false`. One I see is running custom soc descriptor on some different silicon, but there custom soc descriptor could hit harvested rows. Not sure whether that is something we should take care of. No need to do anything about it in this PR, but maybe open an issue so we document this later

Reply:
Yeah I'm currently not sure of the expected use cases